documentation/src/main/asciidoc/topics/search.adoc

```diff
@@ -0,0 +1,1087 @@
+----
+
+==== Selecting attributes
+In some use cases returning the whole domain object is overkill if only a small subset of the attributes are actually
+used by the application, especially if the domain entity has embedded entities. The query language allows you to specify
+a subset of attributes (or attribute paths) to return - the projection. If projections are used then the `QueryResult.list()`
+will not return the whole domain entity but will return a _List_ of _Object[]_, each slot in the array corresponding to
+a projected attribute.
+
+//TODO document what needs to be configured for an attribute to be available for projection.
+
+[source,sql]
+----
+# match all books that have "Data Grid" in their title or description
+# and return only their title and publication year
+SELECT title, publicationYear FROM com.acme.Book WHERE title like '%Data Grid%' OR description like '%Data Grid%'
+----
+
+==== Sorting
+Ordering the results based on one or more attributes or attribute paths is done with the `ORDER BY` clause. If multiple sorting criteria
+are specified, then the order will dictate their precedence.
+
+
+//TODO document what needs to be configured for an attribute to be available for sorting.
+
+[source,sql]
+----
+# match all books that have "Data Grid" in their title or description
+# and return them sorted by the publication year and title
+FROM com.acme.Book WHERE title like '%Data Grid%' ORDER BY publicationYear DESC, title ASC
+----
+
+==== Grouping and Aggregation
+
+{brandname} has the ability to group query results according to a set of grouping fields and construct aggregations of
+the results from each group by applying an aggregation function to the set of values that fall into each group.
+Grouping and aggregation can only be applied to projection queries (queries with one or more field in the SELECT clause).
+
+The supported aggregations are: avg, sum, count, max, min.
+
+The set of grouping fields is specified with the `GROUP BY` clause and the order used for defining grouping fields is
+not relevant. All fields selected in the projection must either be grouping fields
+or else they must be aggregated using one of the grouping functions described below. A projection field can be
+aggregated and used for grouping at the same time. A query that selects only grouping fields but no aggregation fields
+is legal.
+⁠
+Example: Grouping Books by author and counting them.
+[source,sql]
+----
+SELECT author, COUNT(title) FROM com.acme.Book WHERE title LIKE '%engine%' GROUP BY author
+----
+
+NOTE: A projection query in which all selected fields have an aggregation function applied and no fields are used for
+grouping is allowed. In this case the aggregations will be computed globally as if there was a single global group.
+
+==== Aggregations
+
+The following aggregation functions may be applied to a field: avg, sum, count, max, min
```

Comment:
Better to address the reader directly, avoid passive voice, use "can" over "may".

You can apply the following aggregation functions to fields:

IMO it's worth removing "avg, sum, count, max, min" because the sentence introduces the unordered list.

Reply:
You may be sure I will change this :smile: